xarray/tests/test_backends_datatree.py

```diff
@@ -170,6 +208,26 @@ def test_open_datatree(self, unaligned_datatree_nc) -> None:
         ):
             open_datatree(unaligned_datatree_nc)
 
+    @requires_dask
+    def test_open_datatree_chunks(self, tmpdir, simple_datatree) -> None:
+        filepath = tmpdir / "test.nc"
+
+        root_data = xr.Dataset({"a": ("y", [6, 7, 8]), "set0": ("x", [9, 10])})
+        set1_data = xr.Dataset({"a": ("y", [-1, 0, 1]), "b": ("x", [-10, 6])})
+        set2_data = xr.Dataset({"a": ("y", [1, 2, 3]), "b": ("x", [0.1, 0.2])})
+        original_tree = DataTree.from_dict(
+            {
+                "/": root_data.chunk({"x": 2, "y": 1}),
+                "/group1": set1_data.chunk({"x": 1, "y": 2}),
+                "/group2": set2_data.chunk({"x": 2, "y": 3}),
+            }
+        )
+        original_tree.to_netcdf(filepath, engine="netcdf4")
+
+        with open_datatree(filepath, engine="netcdf4", chunks={}) as tree:
+            xr.testing.assert_identical(tree, original_tree)
+            assert_chunks_equal(tree, original_tree, enforce_dask=True)
```

Comment:
for some reason that still eludes me, this:
```python
xr.Dataset({"a": ("x", range(100))}).chunk({"x": 10}).to_netcdf("test.nc", mode="w")
```
writes a single chunk to disk.

This can be worked around by setting the `chunksizes` encoding per variable, but it feels a bit weird that that would be necessary when `zarr` keeps the chunking structure in memory unless told otherwise (and also for `DataTree` objects that becomes confusing very quickly with the number of nodes). Does anyone know what I'm doing wrong?

Reply:
@kmuelbaher any ideas?

Reply:
Using netCDF chunks implies using netCDF4's compression, which I have never found to be very useful (and may be one reason why it is not turned on by default in Xarray). The problem is that netCDF4's compression uses gzip, which is painfully slow to read and write. I would usually rather have unchunked/uncompressed data, which netCDF can read efficiently using arbitrary access patterns.

Reply:
Importantly, note that unchunked/uncompressed netCDF data does not require that an entire chunk is read/written at a time (which is not the case for Zarr). So it still works quite well for Dask.

Reply:
I think the original choice to have Dask chunks map to Zarr chunks by default was bad too. The chunk sizes that dask likes are too large for I/O. It would be good to avoid that mistake here.

Reply:
These comments are interesting but I'm still unclear what @keewis and I should do in the short term to be able to fix this test and merge this PR?

Reply:
> I think the original choice to have Dask chunks map to Zarr chunks by default was bad too. The chunk sizes that dask likes are too large for I/O. It would be good to avoid that mistake here.

Indeed. Perhaps we should have an explicitly `chunks` argument in `to_zarr()`, as more usable (and future proof) alternative to setting `encoding`.

Reply:
> These comments are interesting but I'm still unclear what @keewis and I should do in the short term to be able to fix this test and merge this PR?

I would explicitly set `chunks` here in `open_datatree()`, to verify that they get set properly when reading.

Separately, I would verify that `chunks` are written to Zarr properly. This will require inspecting `encoding` (or using raw Zarr APIs), something like https://github.com/google/xarray-beam/blob/f5a40dbf63661f75d206a260dc997ea64b12069c/xarray_beam/_src/zarr.py#L47

Reply:
I guess this means I should give up trying to write tests for a feature that I decided to punt to a different PR and use uniform chunksizes instead :sweat_smile:

Also I'd like to not modify the `to_zarr` / `to_netcdf` tests in this PR; I've been using those since other tests did so already, but only to create the files, not to test those methods... I guess a cleaner version would be to create the files directly using `zarr` / `h5netcdf` / `netcdf4`.